pkg/descheduler/evictions/evictions.go

```diff
@@ -103,56 +102,75 @@ func (pe *PodEvictor) TotalEvicted() uint {
 	return total
 }
 
-// EvictPod returns non-nil error only when evicting a pod on a node is not
-// possible (due to maxPodsToEvictPerNode constraint). Success is true when the pod
-// is evicted on the server side.
-func (pe *PodEvictor) EvictPod(ctx context.Context, pod *v1.Pod, node *v1.Node, strategy string, reasons ...string) (bool, error) {
-	reason := strategy
-	if len(reasons) > 0 {
-		reason += " (" + strings.Join(reasons, ", ") + ")"
+// NodeLimitExceeded checks if the number of evictions for a node was exceeded
+func (pe *PodEvictor) NodeLimitExceeded(node *v1.Node) bool {
```

Comment:
it seems that NodeLimitExceeded don't calcuate time, if we run descheduler long enough, then NodeLimitExceeded it seems must be true, please correct me if i am wrong.

Reply:
We initialize a new PodEvictor for each iteration/run.